addons/microsoft_calendar/models/calendar.py

```diff
@@ -347,6 +347,23 @@ def _microsoft_values(self, fields_to_sync, initial_values={}):
 
         return values
 
+    def _ensure_attendees_have_email(self):
+        if self.env.context.get("_attendees_have_email", False):
+            return
+        invalid_events = ['\t- %s: %s' % (event.display_time, event.display_name)
+                          for event in self if any(not a.email for a in event.attendee_ids)]
+        if invalid_events:
+            invalid_events.sort()
+            list_length_limit = 50
+            limited_names_list = '\n'.join(invalid_events[:list_length_limit])
+            details = "(%d/%d)" % (list_length_limit, len(invalid_events)) if list_length_limit < len(invalid_events) else "(%d)" % len(invalid_events)
+            raise ValidationError(_("For a correct synchronization between Odoo and Outlook Calendar, "
+                                    "all attendees must have an email address."
+                                    "\nHowever, some events do not respect this condition %s:"
+                                    "\n%s"
+                                    "\nAs long as the events are incorrect, the calendars will not be synchronized. "
+                                    "Either update the events/attendees or archive these events.", details, limited_names_list))
+
```

Comment:
Did you checked how it is rendered with 50  events on small screen? If the list of event is cropped and the bottom message is not displayed, I would put the second part on top of the list. So that you have:

```
For a correct synchronization between Odoo and Outlook Calendar, all attendees must have an email address.
However, some events do not respect this condition %(details)s.
As long as the events are incorrect, the calendars will not be synchronized.
Either update the events/attendees or archive these events.
%(event_list_limited)s
```

So if the list is cropped on a mobile, the user will have the whole message before the list of events. He understands that he must start with the firsts and get the information that nothing will be done before that.
If your current code renders correctly in mobile, don't bother.

Reply:
I checked on mobile and the list is not cropped, but I prefer your suggestion so I moved the list at the end of the warning.